hwmv2: espressif: port all esp32-based boards

nordicjm opened the pull request:
This soc was not created properly, and the original board file which was then cloned for all the other boards was wrong too and should never have been merged

Port needs finishing (feel free to continue): the cpu clusters in the soc need adding and setup properly, boards need to select them properly, and esp32sc3 needs to have the soc parts done properly

nordicjm committed:
soc: espressif_esp32: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: esp32c3_devkitm: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: esp32c3_luatos_core: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: stamp_c3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: xiao_esp32c3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: riscv: icev_wireless: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

sylvioalves commented:
@nordicjm You mean this SoC was not properly create in this collab-hwm branch or also in main branch?

nordicjm commented:
> @nordicjm You mean this SoC was not properly created in this collab-hwm branch or also in main branch?

`main`, there is a soc series which boards are selecting, no actual soc entry and instead that's left as a part number for which one board doesn't even select

sylvioalves commented:
@nordicjm Will you submit a PR to main as well?

nordicjm committed:
boards: xtensa: esp32_devkitc_wroom: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp32_devkitc_wrover: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp32_ethernet_kit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp_wrover_kit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp32s2_franzininho: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp32s2_lolin_mini: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp32s2_saola: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xtensa: esp32s3_devkitm: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
> @nordicjm Will you submit a PR to main as well?

No, it needs to be tidied up here and ready for merging as part of hwmv2

sylvioalves commented:
I've seen you also updated xtensa's ESP32 boards to V2 model. Let me know if I can help and/or if there will be pending changes about it. Thanks.

nordicjm commented:
@sylvioalves what is going on with this? Why are there 2 Kconfigs to select which core is being used that seemingly don't need either to be selected https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/xtensa/espressif_esp32/esp32/Kconfig.series#L21 e.g. for `esp32_devkitc_wroom`

sylvioalves commented:
> @sylvioalves what is going on with this? Why are there 2 Kconfigs to select which core is being used that seemingly don't need either to be selected https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/xtensa/espressif_esp32/esp32/Kconfig.series#L21 e.g. for `esp32_devkitc_wroom`

This is used in AMP use case. This is how we select which core is used for any of the AMP applications.

sylvioalves committed:
boards: xtensa: heltec_wifi_lora32_v2: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: heltec_wireless_stick_lite_v3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: kincony_kc868_a32: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: m5stack_atoms3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: m5stack_atoms3_lite: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: m5stack_core2: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: m5stack_stamps3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: m5stickc_plus: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: odroid_go: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: olimex_esp32_evb: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: xiao_esp32s3: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: xtensa: yd_esp32: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves commented:
I will update this in a couple of hours.

sylvioalves committed:
boards: xtensa: esp32s3_luatos_core: Convert to v2

Converts the board to hwmv2

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

nordicjm commented on boards/espressif/esp32_devkitc_wroom/board.yml:
```diff
@@ -0,0 +1,9 @@
+boards:
+  - name: esp32_devkitc_wroom
+    vendor: espressif
+    socs:
+      - name: esp32
+  - name: esp32_devkitc_wroom_appcpu
+    vendor: espressif
+    socs:
+      - name: esp32
```
this is not correct

nordicjm commented on boards/espressif/esp32_devkitc_wrover/board.yml:
```diff
@@ -0,0 +1,9 @@
+boards:
+  - name: esp32_devkitc_wrover
+    vendor: espressif
+    socs:
+      - name: esp32
+  - name: esp32_devkitc_wrover_appcpu
+    vendor: espressif
+    socs:
+      - name: esp32
```
this is not correct

nordicjm commented on boards/espressif/esp32s3_devkitm/board.yml:
```diff
@@ -0,0 +1,9 @@
+boards:
+  - name: esp32s3_devkitm
+    vendor: espressif
+    socs:
+      - name: esp32s3
+  - name: esp32s3_devkitm_appcpu
+    vendor: espressif
+    socs:
+      - name: esp32s3
```
this is not correct

nordicjm commented on soc/espressif/soc.yml:
```diff
@@ -0,0 +1,15 @@
+family:
+  - name: espressif_esp32
+    series:
+      - name: esp32
+        socs:
+          - name: esp32
+      - name: esp32s2
+        socs:
+          - name: esp32s2
+      - name: esp32s3
+        socs:
+          - name: esp32s3
+      - name: esp32c3
+        socs:
+          - name: esp32c3
```
this is not correct, one or more of these have multiple CPUs, so why are they not listed?

sylvioalves replied:
I think you are referring to clustering options, right? (appcpu and procpu instead)

nordicjm replied:
Yes, 2 CPUs or core means it needs to have a `cpucluster`

sylvioalves replied:
@nordicjm Having that split with cpucluster requires more modifications, not only regarding V2 model porting. Keeping that as it has been, i.e, different type of boards, allows AMP/IPM stuff to work: `samples/drivers/ipm/ipm_esp32/`. I suggest we keep this way and create a new PR (which I will) to improve AMP support and allow proper cpu clustering.

sylvioalves replied:
Why not acceptable? What is the issue on having 2 different board variant within the same board? Is having more that 1 board variant in the same board spec wrong?

sylvioalves replied:
I'll add those cpuclusterings and evaluate the whole effort to make this work. I'll ping back soon.

sylvioalves commented:
@nordicjm Will be there a new PR fixing `MAINTAINERS.yml` fixing folder structure or can I just fix in this PR?

nordicjm commented:
> @nordicjm Will be there a new PR fixing `MAINTAINERS.yml` folder structure or can I just fix in this PR what is esp32-related?
> 
> ```
> -- Generating done
> -- Build files have been written to: /runner/_work/zephyr/zephyr/doc/_build
> cmake --build _build --target html
> [0/2] Generating Devicetree bindings documentation...
> [1/2] Running Sphinx HTML build...
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/xenvm/' in 'files' in area 'Xen Platform' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/gd_gd32/' in 'files' in area 'GD32 Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/silabs_*/' in 'files' in area 'SiLabs Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/ef*/' in 'files' in area 'SiLabs Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/mimx*/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/frdm*/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/lpcxpress*/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/twr_*/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/vmu*/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/nxp_imx/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/nxp_kinetis/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/nxp_lpc/' in 'files' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/*s32*/' in 'files-exclude' in area 'NXP Platforms (MCUX)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/s32*/' in 'files' in area 'NXP Platforms (S32)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/mr_canhubk3/' in 'files' in area 'NXP Platforms (S32)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/arm/ucans32k1sic/' in 'files' in area 'NXP Platforms (S32)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/nxp_s32/' in 'files' in area 'NXP Platforms (S32)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/xtensa/nxp_adsp/' in 'files' in area 'NXP Platforms (Xtensa)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/xtensa/nxp_adsp_*/' in 'files' in area 'NXP Platforms (Xtensa)' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/xtensa/esp32*/' in 'files' in area 'Espressif Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/xtensa/espressif_esp32*/' in 'files' in area 'Espressif Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/riscv/esp32*/' in 'files' in area 'Espressif Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/riscv/espressif_esp32*/' in 'files' in area 'Espressif Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/xilinx*/' in 'files' in area 'Xilinx Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'boards/cypress/xmc*_relax*/' in 'files' in area 'Infineon Platforms' does not match any files
> /runner/_work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'soc/arm/cypress/' in 'files' in area 'Infineon Platforms' does not match any files
> /runner/_work/zephyr/zephyr/doc/_build/src/boards/boards_legacy/xtensa/index.rst:6: WARNING: toctree glob pattern '**/index' didn't match any documents
> ```

Maintainers file can be updated here, the error there is likely because the xtensa directory is now empty and can be removed along with listing it in the index file

sylvioalves committed:
MAINTAINERS: update Espressif entries

Make sure there is no invalid folder.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: legacy: remove index entry for xtensa/riscv boards.

There are no documents to be build within this
board_legacy folder.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
samples: ipm_esp32: fix board labels and skip testing

This sample code is currently not yet read for HWMv2 and sysbuild.
Skip this test for now and have a fix later.

Make sure sample code uses HWMv2 for board
naming convention.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves commented:
@nordicjm, would you take a look in reference board `esp32s3_devkitm` regarding cpucluster and board definitions?  
https://github.com/zephyrproject-rtos/zephyr/pull/69194/commits/d4c2e17cba1f1f23334983af9de842f60b135dfd

sylvioalves commented on soc/espressif/esp32s3/Kconfig.soc:
```diff
@@ -0,0 +1,157 @@
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
+
+config SOC_ENABLE_APPCPU
```
@nordicjm, I had to create this new Kconfig in order to let build system to indicate whether appcpu should be triggered and used in an AMP/Dual-core scenario. This is a bit similar to `config SECOND_CORE_MCUX`. Any considerations?

nordicjm replied:
If it just starts the other core running then that's fine

nordicjm commented on soc/espressif/soc.yml:
```diff
@@ -0,0 +1,21 @@
+    socs:
+    - name: esp32
+      cpuclusters:
+      - name: procpu
+      - name: appcpu
+  - name: esp32s2
+    socs:
+    - name: esp32s2
+  - name: esp32s3
+    socs:
+    - name: esp32s3
+      cpuclusters:
+      - name: procpu
+      - name: appcpu
```
cpu first i.e. `cpupro` and `cpuapp`

sylvioalves replied:
Does Zephyr build system require cpu prefix position somehow?

sylvioalves replied:
I ask this because current naming is based on ESP documentation, i.e., cpu as suffix.

sylvioalves replied:
Ok if I create a top commit changing this in all files (instead of manually editing commit by commit)?

nordicjm replied:
yes that's fine

nordicjm commented on soc/espressif/esp32/CMakeLists.txt:
```diff
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if (CONFIG_SOC_ESP32_APPCPU)
+if(CONFIG_SOC_ESP32_APPCPU)
```
these would need to change then too i.e. `CONFIG_SOC_ESP32_CPUAPP` (use `sed` for global search and replace)

nordicjm commented on soc/espressif/esp32/Kconfig.soc:
```diff
@@ -0,0 +1,164 @@
+	help
+	  ESP32_WROVER_E_N16R8
+
+config SOC_ESP32_PROCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for PROCPU (core 0).
+
+config SOC_ESP32_APPCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
```
rename and move to `Kconfig`

sylvioalves replied:
@nordicjm moving this symbols to Kconfig will fail in board level during compliance check:  
```c
warning: undefined symbol SOC_ESP32S3_PROCPU:

- Referenced at boards/espressif/esp32s3_devkitm/Kconfig.esp32s3_devkitm:4:

config BOARD_ESP32S3_DEVKITM
	bool
	select SOC_ESP32S3_MINI_N8
	select SOC_ESP32S3_PROCPU if BOARD_ESP32S3_DEVKITM_ESP32S3_PROCPU
	select SOC_ESP32S3_APPCPU if BOARD_ESP32S3_DEVKITM_ESP32S3_APPCPU
warning: undefined symbol SOC_ESP32S3_APPCPU:

```

nordicjm replied:
They should be selected in the board's `Kconfig` file not `Kconfig.<board>` file, also the `bool` line vanishes

sylvioalves replied:
OK, moving into Kconfig.defconfig in its proper board if solved it.

nordicjm commented on soc/espressif/esp32/Kconfig.soc:
```diff
@@ -0,0 +1,164 @@
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
+
+config SOC_ENABLE_APPCPU
+	bool
+	default y
+	depends on IPM && SOC_ESP32_PROCPU
+	help
+	  This hidden configuration lets PROCPU core to map and start APPCPU whenever IPM is enabled.
```
`Kconfig`

nordicjm commented on soc/espressif/esp32/Kconfig.soc:
```diff
@@ -0,0 +1,164 @@
+	default "ESP32_WROVER_E_N16R8" if SOC_ESP32_WROVER_E_N16R8
+
+config SOC_TOOLCHAIN_NAME
+	default "espressif_esp32" if SOC_SERIES_ESP32
```
needs a `string` type here (sorry might have asked to remove it before thinking it was global but it is not)

nordicjm commented on soc/espressif/esp32s2/Kconfig.soc:
```diff
@@ -0,0 +1,126 @@
+	default "ESP32S2_WROVER_N16R2" if SOC_ESP32S2_WROVER_N16R2
+
+config SOC_TOOLCHAIN_NAME
+	default "espressif_esp32s2" if SOC_SERIES_ESP32S2
```
as above

nordicjm commented on soc/espressif/esp32s3/Kconfig.soc:
```diff
@@ -0,0 +1,157 @@
+	default "ESP32S3_WROOM_N16R2" if SOC_ESP32S3_WROOM_N16R2
+
+config SOC_TOOLCHAIN_NAME
+	default "espressif_esp32s3" if SOC_SERIES_ESP32S3
```
as above

nordicjm commented on soc/espressif/esp32s3/Kconfig.soc:
```diff
@@ -0,0 +1,157 @@
+	help
+	  ESP32S3_WROOM_N16R2
+
+config SOC_ESP32S3_PROCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for PROCPU (core 0).
+
+config SOC_ESP32S3_APPCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
+
+config SOC_ENABLE_APPCPU
```
`Kconfig`

nordicjm commented on samples/drivers/ipm/ipm_esp32/CMakeLists.txt:
```diff
@@ -4,12 +4,12 @@ cmake_minimum_required(VERSION 3.20.0)
-  set(BOARD_REMOTE "esp32_devkitc_wroom_appcpu")
-elseif("${BOARD}" STREQUAL "esp32s3_devkitm")
-  set(BOARD_REMOTE "esp32s3_devkitm_appcpu")
+if("${BOARD}" STREQUAL "esp32_devkitc_wrover/esp32/procpu")
+  set(BOARD_REMOTE "esp32_devkitc_wrover/esp32/appcpu")
+elseif("${BOARD}" STREQUAL "esp32_devkitc_wroom/esp32/procpu")
+  set(BOARD_REMOTE "esp32_devkitc_wroom/esp32/appcpu")
+elseif("${BOARD}" STREQUAL "esp32s3_devkitm/esp32s3/procpu")
```
This sample should be updated to use sysbuild because it is wrong but that is a future task, as for here the if() names don't look to have been updated and are still using Xcpu

sylvioalves replied:
I have not yet updated cpu prefix. Anyway, I'll update the sample code to use sysbuild in a new PR.

nordicjm commented on boards/espressif/esp32_devkitc_wroom/esp32_devkitc_wroom_esp32_appcpu.dts:
```diff
@@ -0,0 +1,74 @@
+			reg = <0x00110000 0x00100000>;
+		};
+
+		/* Reserve 256kB for the scratch partition */
+		scratch_partition: partition@210000 {
+			   label = "image-scratch";
+			   reg = <0x00210000 0x00040000>;
+		};
```
Completely unrelated to the changes here or hwmv2 but MCUboot swap using scratch mode should not be used if devices can use swap using mode (needs same sector sizes for both slots) which looks to be the case here - might be worth changing in future after hwmv2 is finished

sylvioalves replied:
If it is ok, I'd rather make this changes (sysbuild and flash map) in other PR. I added this snippet in all those .dts file to make sure build and `ipm_esp32` works.

nordicjm commented on boards/espressif/esp32_devkitc_wrover/Kconfig.defconfig:
```diff
@@ -1,10 +1,10 @@
 
-config BOARD
-	default "esp32_devkitc_wrover"
+config SOC_ESP32_PROCPU
```
As above

nordicjm commented on boards/espressif/esp32_devkitc_wrover/Kconfig.defconfig:
```diff
@@ -17,19 +17,17 @@ choice BT_HCI_BUS_TYPE
 
-config BOARD
-	default "esp32_devkitc_wrover_appcpu"
+config SOC_ESP32_APPCPU
+	default y
```
As above

nordicjm commented on boards/espressif/esp32_ethernet_kit/Kconfig.defconfig:
```diff
@@ -3,9 +3,10 @@
-	depends on BOARD_ESP32_ETHERNET_KIT
+if BOARD_ESP32_ETHERNET_KIT_ESP32_PROCPU
+
+config SOC_ESP32_PROCPU
+	default y
```
and for all boards, etc.

nordicjm commented on boards/espressif/esp32_ethernet_kit/Kconfig.defconfig:
```diff
@@ -27,3 +25,18 @@ config HEAP_MEM_POOL_ADD_SIZE_BOARD
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	default 4096
+
+config KERNEL_BIN_NAME
+	default "esp32_appcpu_firmware"
```
why is this being set?

sylvioalves replied:
This comes from V1 model. Sample `ipm_esp32` requires `remote` build to be added in main sample as .c source:  
```c
set_source_files_properties(${REMOTE_ZEPHYR_DIR}/esp32_appcpu_firmware.c PROPERTIES GENERATED TRUE)
target_sources(app PRIVATE src/main.c ${REMOTE_ZEPHYR_DIR}/esp32_appcpu_firmware.c)
```
That sample needs to be converted to sysbuild. Can we keep as is for now?

nordicjm replied:
This needs to go, neither builds or specific boards should be including a .c file, it can be added back in once it is properly using sysbuild. No board with this is going to pass CI

sylvioalves replied:
Then that sample will not work until it is updated. Are you ok with that?

nordicjm commented on boards/espressif/esp_wrover_kit/Kconfig.defconfig:
```diff
@@ -20,3 +21,18 @@ endchoice
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	default 4096
+
+config KERNEL_BIN_NAME
+	default "esp32_appcpu_firmware"
```
?? This should not be being set by any board

nordicjm commented on boards/m5stack/m5stack_atoms3_lite/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: m5stack_atoms3_lite
```
this board should just be a variant of the original since it's the same but without a display or whatnot

sylvioalves replied:
This would also require updating documentation to join this board and its variant details? Or it can be still split?

sylvioalves replied:
Ok, just add an index.rst and add documentation files accordingly.

sylvioalves replied:
Ok, I'd rather not perform this change right now. Board is in place and will required more work. I'll check other pending issues so we can get everything in place.

sylvioalves commented on boards/luatos/esp32c3_luatos_core/esp32c3_luatos_core_esp32c3_usb.yaml:
```diff
@@ -1,4 +1,4 @@
-identifier: esp32c3_luatos_core_usb
+identifier: esp32c3_luatos_core/esp32c3/usb
```
Of course!

sylvioalves commented on boards/espressif/esp32_devkitc_wroom/Kconfig.defconfig:
```diff
@@ -19,20 +19,17 @@ choice BT_HCI_BUS_TYPE
 
-config BOARD
-	default "esp32_devkitc_wroom_appcpu"
+config SOC_ESP32_APPCPU
+	default y
```
I don´t  get it. How to select this in `Kconfig' level if this Kconfig is linked to which cpucluster is used?  
```c
if BOARD_ESP32_DEVKITC_WROOM_ESP32_PROCPU

config SOC_ESP32_PROCPU
	default y
```

sylvioalves committed:
vendors: update vendors lists

Add esp32-based board vendors.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
boards: shields: m5stack_core2_ext: update board name

Update documentation to meet HWMv2 naming convention.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
samples: tests: remove platform_exclude of esp32 boards

Those related tests are automatically filtered out and
the boards can be removed from there.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
samples: tests: update esp32-based board naming

Make sure all those samples and tests use cpu cluster naming.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves commented on samples/drivers/ipm/ipm_esp32/sample.yaml:
```diff
@@ -2,7 +2,10 @@ sample:
     tags:
       - samples
       - ipm
+    skip: true
```
@nordicjm Skipping this test for now. Will enable again once the sample is converted to sysbuild as required.

sylvioalves commented on samples/arch/smp/pktqueue/sample.yaml:
```diff
@@ -18,13 +18,5 @@ tests:
     depends_on:
       - smp
     filter: (CONFIG_MP_MAX_NUM_CPUS > 1)
-    platform_exclude:
```
In the past this was needed. Now, all this boards are filtered out. No need to rename it to new board model. Instead, delete this.

nordicjm commented:
CI must be passing

nordicjm commented on boards/luatos/esp32s3_luatos_core/Kconfig:
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_ESP32S3_LUATOS_CORE
+	select SOC_ESP32_PROCPU if BOARD_ESP32S3_LUATOS_CORE_ESP32S3_PROCPU
+	select SOC_ESP32_APPCPU if BOARD_ESP32S3_LUATOS_CORE_ESP32S3_APPCPU
+
+config BOARD_ESP32S3_LUATOS_CORE_USB
+	select SOC_ESP32_PROCPU if BOARD_ESP32S3_LUATOS_CORE_USB_ESP32S3_PROCPU
+	select SOC_ESP32_APPCPU if BOARD_ESP32S3_LUATOS_CORE_USB_ESP32S3_APPCPU
```
This causes a problem because for the opposite board it will not be defined, 2 possible solutions:

1. Make the usb a variant of the main board, then you can use the base board name here once instead
2. Keep boards differently then use something like this:
```
config BOARD_ESP32S3_LUATOS_CORE_EXTRA_CPU
	bool
	default y
	depends on BOARD_ESP32S3_LUATOS_CORE || BOARD_ESP32S3_LUATOS_CORE_USB
	select SOC_ESP32_PROCPU if BOARD_ESP32S3_LUATOS_CORE_ESP32S3_PROCPU || BOARD_ESP32S3_LUATOS_CORE_USB_ESP32S3_PROCPU
	select SOC_ESP32_APPCPU if BOARD_ESP32S3_LUATOS_CORE_ESP32S3_APPCPU || BOARD_ESP32S3_LUATOS_CORE_USB_ESP32S3_APPCPU
```

sylvioalves commented on boards/franzininho/esp32s2_franzininho/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: esp32s2_franzininho
+  vendor: espressif
```
missed updating, will do so.

sylvioalves commented on soc/espressif/esp32/Kconfig:
```diff
@@ -1,60 +1,42 @@
+	select HAS_PM
+	select HAS_POWEROFF
+
+config SOC_ESP32_PROCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for PROCPU (core 0).
+
+config SOC_ESP32_APPCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
+
+config SOC_ENABLE_APPCPU
+	bool
+	default y
+	depends on IPM && SOC_ESP32_PROCPU
+	help
+	  This hidden configuration lets PROCPU core to map and start APPCPU whenever IPM is enabled.
```
Will update in this PR.

tejlmand commented on boards/luatos/esp32c3_luatos_core/board.yml:
```diff
@@ -0,0 +1,9 @@
+boards:
+  - name: esp32c3_luatos_core
+    vendor: luatos
+    socs:
+      - name: esp32c3
+  - name: esp32c3_luatos_core_usb
+    vendor: luatos
+    socs:
+      - name: esp32c3
```
why is this two boards ?

Seems to me this ought to be made like this:
```
boards:
  - name: esp32c3_luatos_core
    vendor: luatos
    socs:
      - name: esp32c3
        variants:
        - name: usb
```

which is also what is documented here:
https://github.com/zephyrproject-rtos/zephyr/blob/2851032b1b46aab33f46fe165de36e98107c6e5b/boards/luatos/esp32c3_luatos_core/doc/index.rst?plain=1#L43

(this comment also applies to the luatos board)

tejlmand replied:
The `esp32c3_luatos_core` is still implemented as two boards, but documented as `esp32c3_luatos_core/esp32c3`  and the `esp32c3_luatos_core/esp32c3/usb`  variant.

tejlmand commented on boards/luatos/esp32s3_luatos_core/Kconfig:
```diff
@@ -0,0 +1,9 @@
+	bool
+	default y
+	depends on BOARD_ESP32S3_LUATOS_CORE || BOARD_ESP32S3_LUATOS_CORE_USB
+	select SOC_ESP32S3_PROCPU if BOARD_ESP32S3_LUATOS_CORE_ESP32S3_PROCPU || BOARD_ESP32S3_LUATOS_CORE_USB_ESP32S3_PROCPU
+	select SOC_ESP32S3_APPCPU if BOARD_ESP32S3_LUATOS_CORE_ESP32S3_APPCPU || BOARD_ESP32S3_LUATOS_CORE_USB_ESP32S3_APPCPU
```
why is this selected here, and not in the `Kconfig.esp32s3_luatos_core` and `Kconfig.esp32s3_luatos_core_usb` files ?

Secondly, if you use two variants instead of two boards, then you can get rid of:
```
	bool
	default y
	depends on BOARD_ESP32S3_LUATOS_CORE || BOARD_ESP32S3_LUATOS_CORE_USB
```

tejlmand replied:
Guess this is indirectly answered here: https://github.com/zephyrproject-rtos/zephyr/pull/69194#discussion_r1507814296

Those symbols are currently not assumed to go to Kconfig.soc tree.

Though the possibility to get rid of 
```
	bool
	default y
	depends on BOARD_ESP32S3_LUATOS_CORE || BOARD_ESP32S3_LUATOS_CORE_USB
```
will still be possible if there is just one board with one variant instead of two boards.

sylvioalves replied:
Fixed. luatos_core uses usb variant model now.

tejlmand commented on boards/luatos/esp32s3_luatos_core/board.yml:
```diff
@@ -0,0 +1,9 @@
+boards:
+  - name: esp32s3_luatos_core
```
same comment regarding two boards vs. using a variant.

One reason for variants in HWMv2 are use-cases like this.

tejlmand commented on soc/espressif/esp32/Kconfig:
```diff
@@ -69,6 +34,23 @@ config ESP32_APPCPU_DRAM
 	help
 	  Defines APPCPU DRAM area in bytes.
 
+config SOC_ESP32_PROCPU
```
Why is this here ?
If you want a short name symbol for backwards compatibility, then that is ok, but I believe we should have `SOC_ESP32S3_PROCPU`  and `SOC_ESP32S3_APPCPU` which corresponds to settings in `soc.yml`:
https://github.com/zephyrproject-rtos/zephyr/blob/2851032b1b46aab33f46fe165de36e98107c6e5b/soc/espressif/soc.yml#L15-L18

else it will be hard later to handle the cpu cores in sysbuild, cause sysbuild will only be able to load the board and SoC trees. (`Kconfig.<board>` and `Kconfig.soc` files)

sylvioalves replied:
@tejlmand This symbol `SOC_ESP32_PROCPU` and `SOC_ESP32_APPCPU` are means to tell build system whether it is been build for one core of the other. AMP support requires this in current implementation. So this is basically a CONFIG to select which functions to run or not. We could, eventually, convert it to another naming and remove  `SOC` from there. Thoughts?

sylvioalves replied:
This could be replaced by some CONFIG that defines which cpucluster is being currently used during build. Is that possible?

tejlmand replied:
> @tejlmand This symbol `SOC_ESP32_PROCPU` and `SOC_ESP32_APPCPU` are means to tell build system whether it is been built for one core or the other. AMP support requires this in current implementation. So this is basically a CONFIG to select which functions to run or not. We could, eventually, convert it to another naming and remove `SOC` from there. Thoughts?

Thanks for explaining.
I think for now the symbols can be kept, but I still think having a `config SOC_ESP32_ESP32S3_PROCPU` and  `config SOC_ESP32_ESP32S3_APPCPU` in the Kconfig soc-tree in addition is preferred as this aligns the Kconfig.soc with the information available through the soc.yml (and board.yml).

tejlmand replied:
especially if transitioning to sysbuild for the multi-core / multi-image build, then I assume having this information available in sysbuild will be beneficial.

Note, this thread is not blocking for a +1.
The most important comments for the review are:
https://github.com/zephyrproject-rtos/zephyr/pull/69194#discussion_r1507557481
https://github.com/zephyrproject-rtos/zephyr/pull/69194#discussion_r1507730859

sylvioalves replied:
@tejlmand  
1) Why `config SOC_ESP32_ESP32S3..`? I mean, why repeating multiple soc name?  
2) Those definitions are already in soc-tree Kconfig:  
`soc/espressif/esp32/Kconfig`
`soc/espressif/esp32s3/Kconfig`

tejlmand replied:
> I mean, why repeating multiple soc name?

My bad, seems I got an `ESP32` too much there. Updated ^^^

> Those definitions are already in soc-tree Kconfig:

Yes, in the part that is purely Zephyr Kconfig related.
One part of the many changes in HWMv2, is to provide an self-contained Kconfig tree for the (arch), board (`Kconfig.<board>`) and SoCs (`Kconfig.soc`).
Meaning that you can source the pure Kconfig tree for the hardware model outside of Zephyr.
At first in sysbuild, but in future also for other purposes, like CI / compliance.

If you try to source legacy hardware model Kconfig tree in sysbuild, then you'll get a huge bunch of undefined symbols.
It's simply not possible to do so. It's all or nothing/

HWMv2 provides a self-contained and re-usable Kconfig tree (and that was tree was I referred to as the Kconfig soc-tree). 

> Those definitions are already in soc-tree Kconfig:

and with slight difference that they are without `S3`, so I'm unsure if those symbols are a bit more esp generic, so they cover any (potential) multi-core esp32 family SoC, or if they are specific for the `esp32s3` SoC.

tejlmand reviewed (changes requested):
Why are the Luakos `esp32s3_luatos_core` and `esp32c3_luatos_core` boards done as two boards each (4 in total), and not as usb variants of the general board.

afaict the usb is actually just a variant and for the esp32s3 even documented as such.

Sorry if this has been answered elsewhere, there is quite a long list of comments.

tejlmand commented on soc/espressif/esp32c3/Kconfig.soc:
```diff
@@ -0,0 +1,51 @@
+	help
+	  ESP32C3_FX4
+
+config SOC_ESP32C3_MINI_N4
+	bool
+	select SOC_SERIES_ESP32C3
```
same comment applies to other SoCs in this PR.

tejlmand replied:
Note, if one sees `socs: - name: esp32c3` in a board.yml, then one will assume that
```
if(CONFIG_SOC_ESP32C3)
```
evaluates to true, but that is currently not the case.

So either `SOC_ESP32C3` should be selected in addition to `SOC_ESP32C3_MINI_N4`, which is fine.
Or the board should be having: `socs:  - name: esp32c3_mini_n4`

sylvioalves replied:
@tejlmand Whenever any SOC_PARTNUMBER is select, i.e., `SOC_ESP32C3_MINI_N4`, it automatically selects `SOC_ESP32S3` SoC:  
```c
config SOC_ESP32C3_MINI_N4
	bool
	select SOC_SERIES_ESP32C3
```
This is valid for all `SOC_PART_NUMBER`. Do you still see any issue in here?

sylvioalves replied:
Wait. I see now. It sets `SOC_SERIES_ESP32C3`, not `SOC_ESP32S3`. Let me check that again.

sylvioalves replied:
Ok, fixed.

tejlmand commented on boards/luatos/esp32s3_luatos_core/board.yml:
```diff
@@ -0,0 +1,10 @@
+board:
+  name: esp32s3_luatos_core
+  vendor: luatos
+  socs:
+  - name: esp32s3
+    variants:
+    - name: usb
+      cpucluster: procpu
+    - name: usb
+      cpucluster: appcpu
```
👍

tejlmand commented on samples/drivers/ipm/ipm_esp32/CMakeLists.txt:
```diff
@@ -4,12 +4,12 @@ cmake_minimum_required(VERSION 3.20.0)
-  set(BOARD_REMOTE "esp32_devkitc_wroom_appcpu")
-elseif("${BOARD}" STREQUAL "esp32s3_devkitm")
-  set(BOARD_REMOTE "esp32s3_devkitm_appcpu")
+if("${BOARD}" STREQUAL "esp32_devkitc_wrover/esp32/procpu")
```
No change requested and I know it will be addressed later.

In general this kind of testing is dangerous because typos or renames are not caught by compliance, whereas a test for `CONFIG_BOARD_<full_identifier>` where the config has a typo / rename would fail compliance.

Secondly the split in board + identifier, where clusters are inherited, means that sysbuild will be able to check for just the CPU cluster, and adjust only the cluster identifier, and leave the rest, board + soc + variant identical, thus any board using esp32 or esp32s3 can be supported with a simple test.

tejlmand reviewed (commented):
Seems you missed the `esp32c3_luatos_core` boards.

Those are done as two boards but documented as one.

nordicjm commented:
@sylvioalves merging this now as-is to make it into hwmv2, please address above issues with new PRs, thanks

sylvioalves commented:
> Seems you missed the `esp32c3_luatos_core` boards.
> 
> Those are done as two boards but documented as one.

https://github.com/zephyrproject-rtos/zephyr/pull/69663
